example/sparse_advection/sparse_advection_package.cpp

```diff
@@ -0,0 +1,242 @@
+#include <string>
+#include <vector>
+
+#include <coordinates/coordinates.hpp>
+#include <parthenon/package.hpp>
+
+#include "defs.hpp"
+#include "interface/sparse_pool.hpp"
+#include "kokkos_abstraction.hpp"
+#include "reconstruct/dc_inline.hpp"
+#include "sparse_advection_package.hpp"
+
+using namespace parthenon::package::prelude;
+
+// *************************************************//
+// define the "physics" package Advect, which      *//
+// includes defining various functions that control*//
+// how parthenon functions and any tasks needed to *//
+// implement the "physics"                         *//
+// *************************************************//
+
+namespace sparse_advection_package {
+using parthenon::UserHistoryOperation;
+
+std::shared_ptr<StateDescriptor> Initialize(ParameterInput *pin) {
+  auto pkg = std::make_shared<StateDescriptor>("sparse_advection_package");
+
+  Real cfl = pin->GetOrAddReal("SparseAdvection", "cfl", 0.45);
+  pkg->AddParam("cfl", cfl);
+  Real refine_tol = pin->GetOrAddReal("SparseAdvection", "refine_tol", 0.3);
+  pkg->AddParam("refine_tol", refine_tol);
+  Real derefine_tol = pin->GetOrAddReal("SparseAdvection", "derefine_tol", 0.03);
+  pkg->AddParam("derefine_tol", derefine_tol);
+
+  pkg->AddParam("init_size", static_cast<Real>(0.1));
+
+  // set starting positions
+  Real pos = 0.8;
+  pkg->AddParam("x0", RealArr_t{pos, -pos, -pos, pos});
+  pkg->AddParam("y0", RealArr_t{pos, pos, -pos, -pos});
+
+  // add velocities, field 0 moves in (-1,-1) direction, 1 in (1,-1), 2 in (1, 1), and 3
+  // in (-1,1)
+  Real speed = pin->GetOrAddReal("SparseAdvection", "speed", 1.0) / sqrt(2.0);
+  pkg->AddParam("vx", RealArr_t{-speed, speed, speed, -speed});
+  pkg->AddParam("vy", RealArr_t{-speed, -speed, speed, speed});
+  pkg->AddParam("vz", RealArr_t{0.0, 0.0, 0.0, 0.0});
+
+  // add sparse field
+  Metadata m({Metadata::Cell, Metadata::Independent, Metadata::WithFluxes,
+              Metadata::FillGhost, Metadata::Sparse},
+             std::vector<int>({1}));
+  parthenon::SparsePool pool("sparse", m);
+
+  for (int sid = 0; sid < NUM_FIELDS; ++sid) {
+    pool.Add(sid);
+  }
+  pkg->AddSparsePool(pool);
+
+  pkg->CheckRefinementBlock = CheckRefinement;
+  pkg->EstimateTimestepBlock = EstimateTimestepBlock;
+
+  return pkg;
+}
+
+AmrTag CheckRefinement(MeshBlockData<Real> *rc) {
+  // refine on advected, for example.  could also be a derived quantity
+  auto pmb = rc->GetBlockPointer();
+  auto pkg = pmb->packages.Get("sparse_advection_package");
+  std::vector<std::string> vars;
+  for (int var = 0; var < NUM_FIELDS; ++var) {
+    vars.push_back("sparse_" + std::to_string(var));
+  }
+  // type is parthenon::VariablePack<CellVariable<Real>>
+  const auto &v = rc->PackVariables(vars);
+
+  IndexRange ib = pmb->cellbounds.GetBoundsI(IndexDomain::entire);
+  IndexRange jb = pmb->cellbounds.GetBoundsJ(IndexDomain::entire);
+  IndexRange kb = pmb->cellbounds.GetBoundsK(IndexDomain::entire);
```

Comment:
```suggestion
  IndexRange ib = pmb->cellbounds.GetBoundsI(IndexDomain::interior);
  IndexRange jb = pmb->cellbounds.GetBoundsJ(IndexDomain::interior);
  IndexRange kb = pmb->cellbounds.GetBoundsK(IndexDomain::interior);
```
I don't think we should check for refinement in the ghost zones.

Reply:
I actually had to revert this change (going back to checking the entire domain). When only using interior, the restart check fails (which compares outputs at the end when run with a single run or restarted in the middle). I didn't dig into why this results in different final results but I noticed that the advection example also uses the entire domain.